pkgs/development/libraries/onnxruntime/default.nix

```diff
@@ -106,15 +115,25 @@ stdenv.mkDerivation rec {
     nlohmann_json
     nsync
     re2
-    microsoft-gsl
+    (microsoft-gsl.overrideAttrs (oa: {
+      patches = [
+        (fetchurl {
+          url = "https://raw.githubusercontent.com/microsoft/onnxruntime/v${version}/cmake/patches/gsl/1064.patch";
+          sha256 = "sha256-0kv4z68mBuwrVe7UjTBbe0k7c1EAqm80JFaHrrPm3U0=";
+        })
+      ];
+    }))
   ] ++ lib.optionals pythonSupport [
     python3Packages.numpy
     python3Packages.pybind11
     python3Packages.packaging
   ] ++ lib.optionals stdenv.isDarwin [
     Foundation
     iconv
-  ];
+  ] ++ lib.optionals cudaSupport (with cudaPackages; [
+    cudatoolkit
```

Comment:
`cudatoolkit` is slated for eventual deprecation. Instead we recommend using the "redist" packages: `cudaPackages.cuda_*`. For an example, see [the bitsandbytes derivation](https://github.com/NixOS/nixpkgs/blob/d48c7318283b64de6a41b2c68e28fc292a755ca0/pkgs/development/python-modules/bitsandbytes/default.nix).